liquidator/liquidator.js

```diff
@@ -133,8 +133,8 @@ class Liquidator {
     // The `price` is a BN that is used to determine if a position is liquidatable. The higher the
     // `price` value, the more collateral that the position is required to have to be correctly collateralized.
     // Therefore, we add a buffer by deriving scaledPrice = price * (1 - crThreshold)
-    const scaledPrice = this.fromWei(
-      price.mul(this.toBN(this.toWei("1")).sub(this.toBN(this.toWei(this.crThreshold.toString()))))
+    const scaledPrice = Math.round(
```

Comment:
Not entirely sure how this happened, but with USDETH prices that are < 0 (e.g. 0.004x), `scaledPrice` was returning a string with decimals

Reply:
Is this price supposed to be pre-multiplied be 1e18? Put another way: if the scaled price were representing 0.7, it would be 7e17, right? If so, we probably want to avoid representing it as a js number -- they can lose precision with numbers this large.

Reply:
I know this was here before but why do we need to run the `fromWei` call here actually? later on line 145 we then convert it back to a big number with `this.toBN(scaledPrice)`? would it not make sense to leave this as a big number throughout? seems like it makes it unnecessarily complicated at this point.

Reply:
@mrice32 yes the prices are meant to be pre-multiplied by 1e18: 0.7 scaledPrice should be 7e17. What's going on right now is that all of the math is done in BigNumber format, and then at the end the product is `fromWei()`'d into a string, which I want to remove decimals from. Hence why I wrapped it with `Math.round()`. Can you think of a better way? Unfortunately it does not seem that BN.js has a .round method.

@chrismaree The `fromWei()` actually moves the decimal place left by 18 places. I agree it doesn't have to be a string after this call, but all of the downstream calls expect it to be in a non wei form. In other words, 0.7 should be 0.7, not 7e17

Reply:
Instead of `Math.round()` I could also use String manipulation and `.trim` the decimal point out.

Reply:
@nicholaspai I think you can do it with just BNs:
1. BNs only represent integers. If you try to call `toBN("0.7")`, it will fail. We'll need to avoid using `fromWei()` to do the division since it will produce non-integers.
2. When dividing, `BN`s truncate. So a truncated replacement for `fromWei(input)` using on BNs would be `toBN(input).div(toWei("1"))`.
3. If you'd like to round instead of truncating, you can use the `.divRound()` method instead of `.div()`.

I think that should work, but let me know if I've made any bad assumptions.

Reply:
I made this change [here](https://github.com/UMAprotocol/protocol/pull/1739#discussion_r455964849)